Plugin/Catalog/Helper/Product/Configuration/AddVisibleInCheckoutAttributesToCustomOptionsPlugin.php

```diff
@@ -42,8 +43,34 @@ public function __construct(GetVisibleCheckoutAttributesServiceInterface $getVis
     public function afterGetCustomOptions(Configuration $configuration, array $customOptions, ItemInterface $item)
     {
         $attributes = $this->getVisibleCheckoutAttributesService->execute();
+
+        $configurableAttributes = $item->getOptionByCode('attributes') ? $item->getOptionByCode('attributes')->getValue() : [];
+        $configurableAttributes = $configurableAttributes ? array_keys(json_decode($configurableAttributes, true)) : [];
+
+        $product = $item->getProduct();
+
+        if (!$product) {
+            return [];
+        }
+
         foreach ($attributes as $attribute) {
-            $value = $attribute->getFrontend()->getValue($item->getProduct());
+            $attributeFrontend = $attribute->getFrontend();
+            $value = $attributeFrontend->getValue($product);
+
+            if ($item instanceof AbstractItem && $product->getTypeId() == 'configurable') {
```

Comment:
```suggestion
            if ($item instanceof AbstractItem && $product->getTypeId() === 'configurable') {
```

Reply:
@sprankhub seems that your suggestions are not happening, should we go forward here?